runtime/serde/serde-json/common/test/aws/smithy/kotlin/runtime/serde/json/JsonDeserializerTest.kt

```diff
@@ -553,4 +555,236 @@ class JsonDeserializerTest {
         assertNull(y)
         assertEquals(2, z)
     }
+
+    private fun testDeserializeDocument(doc: String, expected: Document) {
+        // pad the end to verify that the parser token index isn't misaligned after handling the document field
+        val payload = """
+         {
+             "SerializedDocument": $doc,
+             "i": 1
+         }
+        """.trimIndent().encodeToByteArray()
+
+        val deserializer = JsonDeserializer(payload)
+        val serializedDocumentField = SdkFieldDescriptor(SerialKind.Document, JsonSerialName("SerializedDocument"))
+        val iField = SdkFieldDescriptor(SerialKind.Integer, JsonSerialName("i"))
+        val struct = SdkObjectDescriptor.build {
+            field(serializedDocumentField)
+            field(iField)
+        }
+
+        var actual: Document? = null
+        var i: Int? = null
+        deserializer.deserializeStruct(struct) {
+            loop@ while (true) {
+                when (findNextFieldIndex()) {
+                    serializedDocumentField.index -> actual = deserializeDocument()
+                    iField.index -> i = deserializeInt()
+                    null -> break@loop
+                }
+            }
+        }
+
+        if (actual == null) {
+            fail("deserialized document should not be null")
+        }
+        assertDocumentEquals(expected, actual!!)
+        assertEquals(1, i)
+    }
+
+    @Test
+    fun canDeserializeDocumentDouble() =
```

Comment:
we have been wildly inconsistent with test naming conventions, we should probably converge on one (or at least not add a new one). 

The other tests in this file use `it<Name>` style (e.g. `itDoesFoo()`). My vote would be `test<Name>` (e.g. `testDeserializeDocumentDouble()`) which is what I think we have the most of overall in the codebase.

Reply:
I would say either `testFoo()` or full sentences for function names wrapped in backticks, but yeah, I've just been following whatever is in the current file.

Reply:
We shouldn't use the full sentences in backticks, there are issues with KMP (JS I believe) where we have to convert those function names. Kyle ran into this in an early prototype of JS support (we never merged it which is why there are still names like that but presumably we'll have to fix those at some point)